splice/static/js/controllers/authoring.js

```diff
@@ -98,9 +98,73 @@ angular.module('spliceApp').controller('authoringController', function($controll
      * Send tiles to backend for publication.
      * Assumes data is correct.
      */
+    var cloneTile = function (tile) {
+      if (null === tile || "object" !== typeof tile) return tile;
+
+      var copy = tile.constructor();
+      for (var attr in tile) {
+        if (tile.hasOwnProperty(attr)) {
+          if (attr === "imageURI" || attr === "enhancedImageURI") {
+            copy[attr] = "";
+          } else {
+            copy[attr] = tile[attr];
+          }
+        }
+      }
+      return copy;
+    };
+
+    var compressPayload = function (tiles) {
+      /* *
+       * compress the payload for publishing. Note that the tiles might be cached,
+       * therefore we create a cheap (all imageURIs won't be cloned) copy here
+       */
+      var copies = {};
+      var uri2id = {};
+      var assets = {};
+      var id = 0;
+
+      for (var locale in tiles) {
+        if (tiles.hasOwnProperty(locale)) {
```

Comment:
```javascript
for (var locale in tiles) {
...
}
```
This will iterate over the properties of the object `tiles` and store it in a local var `locale`.
The next line is a check whether the property is in the `tiles` object.

If the property `locale` wasn't in `tiles`, line 128 wouldn't  have executed.

Makes sense?